src/Pipeline/Pipeline.php

```diff
@@ -36,13 +36,13 @@ public function run(?callable $observer = null): void
         }
 
         foreach ($this->source->load() as $bucket) {
+            $observer($bucket);
+
             $result = ($this->middlewares)($bucket);
 
             foreach ($result as $resultBucket) {
                 $this->target->save($resultBucket);
             }
-
-            $observer($bucket);
```

Comment:
Do you know why this was executed after the middleware seems this was wrong from the beginning doesnt it ?

Reply:
I guess I didn't think about it. I only noticed it while editing.

Reply:
okey 👍